src/nnet3bin/nnet3-merge-egs.cc

```diff
@@ -105,8 +109,8 @@ int main(int argc, char *argv[]) {
       // Do Next() now, so we can test example_reader.Done() below .
       example_reader.Next();
       num_read++;
-      
-      if (minibatch_ready || (example_reader.Done() && !examples.empty())) {
+
+      if (minibatch_ready || (!discard_partial_minibatches && (example_reader.Done() && !examples.empty()))) {
```

Comment:
Also, you can modify the nnet3 training scripts (TDNN and LSTM) to add the flag --discard-partial-minibatches=true to nnet3-train.  We don't need to make this configurable-- it won't appreciably affect the training, and it will increase speed.